src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -545,43 +545,39 @@ private boolean offerNotEnoughResources(Offer offer, double cpus, int mem) {
   }
 
   private void reconcileTasks(SchedulerDriver driver) {
-    updateReconciliationDeadline();
     driver.reconcileTasks(Collections.<Protos.TaskStatus> emptyList());
-    reconcilePersistentState();
-    reconciliationCompleted = true;
+    Timer timer = new Timer();
+    timer.schedule(new ReconcileStateTask(), conf.getReconciliationTimeout() * 1000);
```

Comment:
For the recommended algorithm, see the "Algorithm" section of http://mesos.apache.org/documentation/latest/reconciliation/